tests/System.IO.Pipelines.Tests/PipeCancellationTokenTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class PipeCancellationTokenTests
```

Comment:
You need tests where you interleave cancellation tokens and `CancelPendingRead` calls.